operations/app_operation_test.go

```diff
@@ -0,0 +1,69 @@
+package operations
+
+import (
+	"bytes"
+	"io"
+)
+
+type MockAppOperationApiClient struct {
+}
+
+func (client MockAppOperationApiClient) Get(path string, body io.Reader) ([]byte, error) {
+	return bytes.NewBufferString("{\"say\":\"hello\"}").Bytes(), nil
+}
+
+func (client MockAppOperationApiClient) Delete(path string, body io.Reader) ([]byte, error) {
+	return bytes.NewBufferString("").Bytes(), nil
+}
+
+type MockAppOperationInputReaderNo struct {
+}
+
+func (_ MockAppOperationInputReaderNo) Read(_ bool) (string, error) {
+	return "n\n", nil
+}
+
+type MockAppOperationInputReaderYes struct {
+}
+
+func (_ MockAppOperationInputReaderYes) Read(_ bool) (string, error) {
+	return "y\n", nil
+}
+
+func ExampleAppOperation_run_delete_no_confirm_output() {
+	client := &MockAppOperationApiClient{}
+	oper := NewAppOperation("asd", Delete, true, client, nil /* because it doesnt matter */)
+
+	oper.run()
```

Comment:
I thought that if the function name started with a lowercase letter, it is a private function.

Reply:
Yes that is correct. In golang the privateness is private within the package. Since the test is in the same package, it has access to private methods.

Reply:
I did not know that 😄  good to learn